### DISCUSSION TITLE
TypeError: ...PathGeneratorFactory::guardAgainstInvalidPathGenerator()

### DISCUSSION THREAD

**Discussion by goyne** [spatie/laravel-medialibrary#2328] [Q&A]
Hi, I'm pulling my hair out. Struggling for days now. From one moment to the next I get the following error: 

TypeError: Spatie\MediaLibrary\Support\PathGenerator\PathGeneratorFactory::guardAgainstInvalidPathGenerator(): Argument #1 ($pathGeneratorClass) must be of type string, null given

The only change is an update from 8 to 9. I have already tried reinstalling the package, same result.

In the config file, the path_generator is indeed set to null, but I cannot find any info in the documentation about what to use here either.

Model is configured as it should be. Furthermore, the media file is added in a fairly basic way:
`$media = $submission->addMediaFromRequest('image')->toMediaCollection('submissions', $disk);`

Any ideas...?

**Comment by freekmurze** [comment]
You could try to debug this by dumping `config('medialibrary')` and verify that it has the values you expect.

**Reply by goyne** [reply]
Yep, solved.

Config file was missing some values. Probably forgot to add them after one of the earlier major updates.
